Support use of subresource alias attrib in substitutions

jimbr70 opened the pull request:
Various and numerous changes. This is the running produciton code for SVL.

ayeleta commented:
Review status: 0 of 14 files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[sandbox_scripts/environment/teardown/teardown_resources.py, line 27 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvUCcKufdHNHOXwfqIT:-KvUCcKufdHNHOXwfqIU:bp2obfr) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/environment/teardown/teardown_resources.py#L27)):*
> ```Python
>         try:
>             if saveNRestoreTool.get_storage_manager():
>                 ignore_models = ['Generic TFTP server', 'Config Set Pool', 'Generic FTP server',
> ```

Remove from here. The ignore_models list is already initiated outside the try

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 110 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYM4-2slOIL9SjRWsc:-KvYM4-2slOIL9SjRWsd:bt8uvy2) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L110)):*
> ```Python
>                         if been_here is False:
>                             been_here = True
>                             self.sandbox.report_info('Pausing 60s before "Validate L2 Routes"', write_to_output_window=True)
> ```

Sleep is set to 90 and not 60

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 114 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYMrO18B0Jn6QEJ35f:-KvYMrO18B0Jn6QEJ35g:bor5wsg) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L114)):*
> ```Python
>                         try:
>                             command_inputs = [InputNameValue('Show_Not_Found_Only', 'y')]
>                             self.sandbox.api_session.ExecuteCommand(self.sandbox.id, resource.name, 'Resource',
> ```

Call resource.execute_command instead

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 147 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYaPzW3Y6dddmodnFW:-KvYaPzW3Y6dddmodnFX:b-am68jf) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L147)):*
>  
We should  handle snapshots with orchestration_restore. Why was that code removed?

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 181 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYXBpojR8vkOGpIGVH:-KvYXBpojR8vkOGpIGVI:b-tv0x5w) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L181)):*
> ```Python
>                     config_path = ''
>                     with lock:
>                         if config_stage == 'Snapshots':
> ```

_get_concrete_config_file _path should return the path in case of a snapshot as well 

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 216 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYcgmaIe39G5TY5za2:-KvYcgmaIe39G5TY5za3:b-81upc2) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L216)):*
> ```Python
>                                 #self.sandbox.report_info("Using image key " + str(image_key) + " for " + resource.name)
>                                 dict_img_version = images_path_dict[image_key].version
>                                 if version < ' ':
> ```

if version != ''

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 223 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYdK3lqq8WGrGpT0di:-KvYdK3lqq8WGrGpT0dj:ba9v7ep) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L223)):*
> ```Python
> 
>                             # same image version - Only load config (running override)
>                             #message += resource.name + ": loading configuration from " + config_path
> ```

Why this is not part of the message?

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 282 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYzcYyfpSOFk_WJoT5:-KvYzcYyfpSOFk_WJoT6:bgye5mt) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L282)):*
> ```Python
>             # If threshold is 0-4, power off now
>             # If threshold is >5, check for usesage in that period with a 2minute shift ahead
>             try:
> ```

Move the power off logic to the resource class. 

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 282 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ-EEmi3jOnO9cWjAX:-KvZ-EEmi3jOnO9cWjAY:ba8f7yt) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L282)):*
>         Why the snapshot part was removed?

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 372 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ-QqlUPzG_XT-I9S-:-KvZ-QqlUPzG_XT-I9S0:b-dczji) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L372)):*
> ```Python
>         # Look for a template config file
>         tmp_template_config_file = tempfile.NamedTemporaryFile(delete=False)
>         tftp_template_config_path = root_path + resource.alias + '_' + resource.model.replace(' ','_') + '.tm'
> ```

No need for the replace. there is a replace statement right after

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 385 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ04vCGGG-GuaJ2MZr:-KvZ04vCGGG-GuaJ2MZs:b-yvsya2) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L385)):*
> ```Python
>                 self.storage_mgr.download(tftp_template_config_path, tmp_template_config_file.name)
>             except Exception as ex:
>                 print ex.message
> ```

Use sandbox.report_info instead

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 392 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ53FxvsqXaC7XRaa_:-KvZ53FxvsqXaC7XRaaa:b-8z76p3) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L392)):*
> ```Python
>         if save_config_for_device:
>             try:
>                 message += '\nSaving configuration for device: ' + resource.name
> ```

Why was the orchestration_save part removed?  We should handle it as well as this is part of product's road map

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 442 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ0g-cV18sdBvn_jk-:-KvZ0g-dG1sgheqa54TU:bgi3f4) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L442)):*
> ```Python
>             if self.storage_mgr.dir_exist(env_dir):
>                 self.sandbox.report_info("Directory for snapshot already exists:\n" + env_dir,
>                                          write_to_output_window=True)
> ```

snapshot directory is for internal use. The user should not be notified on that

---

*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 528 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ6tjM7iJs_a3kUP7G:-KvZ6tjM7iJs_a3kUP7H:b-i0kmsh) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L528)):*
> ```Python
>                 if resource.model.lower() == ignore_model.lower():
>                     return False
> ```

NetworkingSaveNRestore is only handling physical (networking inventory), so all methods should use get_root_networking_resources instead of get_root_resources and the the apps check is not needed

---

*[sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py, line 30 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ9dSTAtIeeEuGgZaD:-KvZ9dSTAtIeeEuGgZaE:b-a9jkp1) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py#L30)):*
> ```Python
> 
>             concrete_config_data = template_config_data
>             subst_log = ""
> ```

This log is only for debug - Either use a flag to set this param or put it as a remark 

---

*[sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py, line 80 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ949osWiMqZY3ac1E:-KvZ949osWiMqZY3ac1F:budghg7) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py#L80)):*
> ```Python
>                 concrete_name = ''
>                 if the_resources is None:  # fetch once the resources
>                     the_resources = sandbox.get_all_resources()
> ```

Why do you get all resources and not just the networking resources? The ConfigFileMAnager only manage networking devices config files

---

*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 31 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZAJniKLKT_Y5HVDtX:-KvZAJniKLKT_Y5HVDtY:b-73899r) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L31)):*
> ```Python
>             #self.model = self.model.replace(' ','_')
>             #Hacked in to cover for filenames too long for product, such as those below:
>             self.model = self.model.replace('Cisco 5500 Series Wireless LAN Controller','WLC')
> ```

Use an if statement for that replace

---

*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 34 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZB-bzjWhK583ec13G:-KvZB-bzjWhK583ec13H:b4c4re1) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L34)):*
> ```Python
>             self.alias = resource_alias
> 
>             self.power_on_wl = ["power_on", "Power On", "Power ON", "PowerOn"]  # whitelist for power on commands
> ```

Save the lists outside of the class and not in the init 

---

*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 118 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZByiCR92ZFvPEsHcw:-KvZByiCR92ZFvPEsHcx:bidzwhz) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L118)):*
> ```Python
> 
>     # -----------------------------------------
>     # -----------------------------------------
> ```

No need to get the resource name as a param. You can take it from self.name

---

*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 243 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZCpwX_WaeFBQbtHIK:-KvZCpwX_WaeFBQbtHIL:b-k3uf07) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L243)):*
> ```Python
>         try:
>             command_inputs = [InputNameValue('folder_path', str(config_path)),
>                                 InputNameValue('configuration_type', str(config_type))]
> ```

Don't remove the old code - This needs to stay for backward compatibility with shell 1st gen older versions

---

*[sandbox_scripts/QualiEnvironmentUtils/Sandbox.py, line 425 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZJOQGD0boBpSbiAgP:-KvZJOQGD0boBpSbiAgQ:b-jrvz6y) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Sandbox.py#L425)):*
> ```Python
>     # -----------------------------------------
>     # -----------------------------------------
>     def update_description(self, description):
> ```

put in a todo remark

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 110 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYM4-2slOIL9SjRWsc:-Kvc1EsTAOur8JrJ2aDQ:b-rux5q4) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L110)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Sleep is set to 90 and not 60
</blockquote></details>

changed message. done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 147 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYaPzW3Y6dddmodnFW:-Kvc43n7Xlcytly1J4NY:b-alk7u7) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L147)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

>  
We should  handle snapshots with orchestration_restore. Why was that code removed?
</blockquote></details>

See email to Costya & Gal - was discussed. Don't remember details - hopefully they will reply

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 223 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYdK3lqq8WGrGpT0di:-Kvc4u4j-mSsVqBe2vYk:b-upvja2) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L223)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Why this is not part of the message?
</blockquote></details>

it is redundant wit the sandbox.report message.  I favored realtime in-process messages versus saving them up. Something we can all discuss a final direction.
Leaving as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 282 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYzcYyfpSOFk_WJoT5:-Kvc5IXuIgidz1A-UroG:bfavprw) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L282)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Move the power off logic to the resource class. 
</blockquote></details>

So this is not something to do with the little time remaining on the account, but I understand your idea. How do we capture this for the 'next roll'?  Insert a Todo?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 282 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ-EEmi3jOnO9cWjAX:-Kvc6Wlglb_OCb7CEU7y:ba8t96z) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L282)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

>         Why the snapshot part was removed?
</blockquote></details>

because I handled it earlier which you caught and said to put in this procedure - it is back, at the top of _get_concrete_config_file_path.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 442 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ0g-cV18sdBvn_jk-:-Kvc7l01KAq-5K2Ws6hV:b31qwxx) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L442)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

snapshot directory is for internal use. The user should not be notified on that
</blockquote></details>

not sure I agree - user may want to access/copy them for whatever... why not?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 392 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ53FxvsqXaC7XRaa_:-Kvc7wPj1u8tFWOY-JMs:b-thk3z2) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L392)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Why was the orchestration_save part removed?  We should handle it as well as this is part of product's road map
</blockquote></details>

Same comment as previous - discussions lab months ago decided to use save/restore. sorry don't recalll all the details.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 528 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ6tjM7iJs_a3kUP7G:-Kvc8iZn2tPTVah1g_wV:b-66k2f) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L528)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

NetworkingSaveNRestore is only handling physical (networking inventory), so all methods should use get_root_networking_resources instead of get_root_resources and the the apps check is not needed
</blockquote></details>

commented out the apps test until I do some tests

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py, line 80 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ949osWiMqZY3ac1E:-Kvc8qSoCMDvEPh1netf:b-15814j) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py#L80)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Why do you get all resources and not just the networking resources? The ConfigFileMAnager only manage networking devices config files
</blockquote></details>

For substitution we have non-networking devices holding attribute values we need

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py, line 30 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ9dSTAtIeeEuGgZaD:-Kvc9S9XkLDVZXAoyCxj:bq7hhrm) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/ConfigFileManager.py#L30)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

This log is only for debug - Either use a flag to set this param or put it as a remark 
</blockquote></details>

commented them all out

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 31 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZAJniKLKT_Y5HVDtX:-Kvc9lYcshr5-aGLUhST:bb76mu4) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L31)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Use an if statement for that replace
</blockquote></details>

Why?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 34 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZB-bzjWhK583ec13G:-KvcA34B4pWJZfru5MyA:bvixxql) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L34)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Save the lists outside of the class and not in the init 
</blockquote></details>

don't understand 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 243 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZCpwX_WaeFBQbtHIK:-KvcBG945t48xBADNn09:b-uuc4lw) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L243)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Don't remove the old code - This needs to stay for backward compatibility with shell 1st gen older versions
</blockquote></details>

? the code is there. don't understand

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 114 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYMrO18B0Jn6QEJ35f:-Kvc2077KT3LyuRmocSB:b-t458zv) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L114)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Call resource.execute_command instead
</blockquote></details>

Done.  But in testing, failed, reverted to orig code.  failed to run for svl290-gg07-c3850-2  ex: 'ResourceBase' object has no attribute 'ExecuteCommand'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 216 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYcgmaIe39G5TY5za2:-Kvc4Z6D9Jtw1swYCwkl:bqmf7es) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L216)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

if version != ''
</blockquote></details>

Done. tried that - fails.  There's some other character or something in there. Reverted. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 181 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYXBpojR8vkOGpIGVH:-Kvc3xlNUSAsKA53GH4-:b-896fix) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L181)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

_get_concrete_config_file _path should return the path in case of a snapshot as well 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 223 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYdK3lqq8WGrGpT0di:-Kvc5CY_JBURFm-t3os4:b-896fix) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L223)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

it is redundant wit the sandbox.report message.  I favored realtime in-process messages versus saving them up. Something we can all discuss a final direction.
Leaving as is.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 372 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ-QqlUPzG_XT-I9S-:-Kvc7DciVK4oJ_oweH0c:b-896fix) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L372)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

No need for the replace. there is a replace statement right after
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/environment/teardown/teardown_resources.py, line 27 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvUCcKufdHNHOXwfqIT:-Kvc16g8zful2GgfqyGE:b-896fix) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/environment/teardown/teardown_resources.py#L27)):*
<details><summary><i>Previously, ayeleta wrote…</i></summary><blockquote>

Remove from here. The ignore_models list is already initiated outside the try
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 147 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYaPzW3Y6dddmodnFW:-KvhbbUqhkF_TWWhCeH-:bgef0sw) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L147)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

See email to Costya & Gal - was discussed. Don't remember details - hopefully they will reply
</blockquote></details>

This will have to be a TODO item due to account engagement timing

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 216 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYcgmaIe39G5TY5za2:-KvhbmIQ1luKzWFNjgKH:bonq33u) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L216)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

Done. tried that - fails.  There's some other character or something in there. Reverted. 
</blockquote></details>

Well, turns out  the if was on the wrong variable!  Fixed test and message.  Nice this was caught.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 114 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYMrO18B0Jn6QEJ35f:-Kvhkp01Hope-N1rYlAl:b-loc4l4) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L114)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

Done.  But in testing, failed, reverted to orig code.  failed to run for svl290-gg07-c3850-2  ex: 'ResourceBase' object has no attribute 'ExecuteCommand'
</blockquote></details>

inserted todo for follow up later

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 147 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYaPzW3Y6dddmodnFW:-KvhmaeyiqPHJETSbi5D:b-do4x98) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L147)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

This will have to be a TODO item due to account engagement timing
</blockquote></details>

will note as todo this has to be high pri change later

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 282 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvYzcYyfpSOFk_WJoT5:-Kvhoy-aaEbLui7rysSb:bj67okr) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L282)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

So this is not something to do with the little time remaining on the account, but I understand your idea. How do we capture this for the 'next roll'?  Insert a Todo?
</blockquote></details>

added todo

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 442 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ0g-cV18sdBvn_jk-:-KvhpqnVMo1i9T2x8SSJ:b-7okwdn) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L442)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

not sure I agree - user may want to access/copy them for whatever... why not?
</blockquote></details>

commented  out

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py, line 392 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZ53FxvsqXaC7XRaa_:-Kvhpw86ENSmsIELVbP-:b7g1g3n) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/e499f30a4fc618b14ad3650613aca6a4d5649e7d/sandbox_scripts/helpers/Networking/NetworkingSaveNRestore.py#L392)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

Same comment as previous - discussions lab months ago decided to use save/restore. sorry don't recalll all the details.
</blockquote></details>

we added a TODO for this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->

jimbr70 commented:
*[sandbox_scripts/QualiEnvironmentUtils/Resource.py, line 34 at r1](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31#-KvZB-bzjWhK583ec13G:-KvhrkmeIrmsJSzGy6L0:b-896fix) ([raw file](https://github.com/qualisystemslab/cloudshell-orch-sandbox/blob/d58c97844ad672065d484576b86039375a9ee9c4/sandbox_scripts/QualiEnvironmentUtils/Resource.py#L34)):*
<details><summary><i>Previously, jimbr70 wrote…</i></summary><blockquote>

don't understand 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystemslab/cloudshell-orch-sandbox/31)*
<!-- Sent from Reviewable.io -->
